src/NLog/LogFactory.cs

```diff
@@ -1092,15 +1093,32 @@ public void ResetCandidateConfigFilePath()
         private static IEnumerable<string> GetDefaultCandidateConfigFilePaths()
         {
             // NLog.config from application directory
-            if (CurrentAppDomain?.BaseDirectory != null)
+            string baseDirectory = PathHelpers.TrimDirectorySeparators(CurrentAppDomain?.BaseDirectory);
+            if (!string.IsNullOrEmpty(baseDirectory))
+                yield return Path.Combine(baseDirectory, "NLog.config");
+
+            bool platformFileSystemCaseInsensitive = PlatformDetector.IsWin32;
+            if (!platformFileSystemCaseInsensitive && !string.IsNullOrEmpty(baseDirectory))
+                yield return Path.Combine(baseDirectory, "nlog.config");
+
+#if !SILVERLIGHT && !NETSTANDARD1_3
+            var entryAssemblyLocation = PathHelpers.TrimDirectorySeparators(AssemblyHelpers.GetAssemblyFileLocation(System.Reflection.Assembly.GetEntryAssembly()));
+            if (!string.IsNullOrEmpty(entryAssemblyLocation))
             {
-                yield return Path.Combine(CurrentAppDomain.BaseDirectory, "NLog.config");
-                yield return Path.Combine(CurrentAppDomain.BaseDirectory, "nlog.config");
+                if (!string.Equals(entryAssemblyLocation, baseDirectory, StringComparison.OrdinalIgnoreCase))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1066)